classes/migration/upgrade/PKPv3_3_0UpgradeMigration.inc.php

```diff
@@ -392,46 +384,41 @@ private function _migrateSubmissionFiles()
                 'submission_file_id' => $row->file_id,
                 'file_id' => $newFileId,
             ]);
-
-            // Update revision data in event logs
-            $eventLogIds = DB::table('event_log_settings')
-                ->where('setting_name', '=', 'fileId')
-                ->where('setting_value', '=', $row->file_id)
-                ->pluck('log_id');
-            DB::table('event_log_settings')
-                ->whereIn('log_id', $eventLogIds)
-                ->where('setting_name', 'fileRevision')
-                ->where('setting_value', '=', $row->revision)
-                ->update(['setting_value' => $newFileId]);
+        }
+        // Set submission event log file IDs to the new IDs where necessary.
+        switch (DB::getDriverName()) {
+            case 'mysql':
+                DB::unprepared("UPDATE event_log_settings elsr JOIN event_log_settings elsf ON (elsr.log_id = elsf.log_id AND elsf.setting_name='fileId') JOIN submission_files sf ON (sf.revision = elsr.setting_value AND sf.file_id = elsf.setting_value) SET elsr.setting_value = sf.new_file_id WHERE elsr.setting_name='fileRevision'");
```

Comment:
@ewhanson, you might need a work-around like this one Jonas added -- I'm about to ping him there for more information. https://github.com/pkp/pkp-lib/pull/7464/files#diff-bc7adf554f1d14817b9725a1aab342442e779c01926fd926f673692d44996408R40

Reply:
This one seems to be ok :)